README.md

```diff
@@ -8,9 +8,221 @@
+
+- **Safer Exceptions**: Using Scala 3's `throws` clauses, functions explicitly declare what exceptions they can throw, enabling better error handling at compile time
+- **Context Functions**: The framework uses `using`/`given` for implicit context passing, making the aggregate computation context available without boilerplate
+- **Type-Level Guarantees**: Strong typing ensures that field operations are correctly aligned and serialization is properly configured
+
+### Programming Style
+
+ScaFi programs are:
+- **Declarative**: Express *what* the system should compute, not *how* each device should behave
+- **Composable**: Build complex behaviors from simple, reusable building blocks
+- **Resilient**: Automatically handle device failures and network topology changes
+- **Pure**: Computation logic is separated from side effects and device-specific concerns
+
+## Installation
+
+### With SBT
+
+Add ScaFi 3 to your `build.sbt`:
+
+```scala
+libraryDependencies ++= Seq(
+  "it.unibo.scafi" %%% "scafi3-core" % "1.0.5",
+  // For distributed systems support
+  "it.unibo.scafi" %%% "scafi3-distributed" % "1.0.5"
+)
+```
+
+ScaFi 3 supports **JVM**, **JavaScript**, and **Native** platforms through Scala's cross-platform compilation:
+
+```scala
+// For a specific platform, use %% instead of %%%
+libraryDependencies += "it.unibo.scafi" %% "scafi3-core" % "1.0.5"
+```
+
+### With Mill
+
+Add ScaFi 3 to your `build.sc`:
+
+```scala
+import mill._, scalalib._
+
+object myproject extends ScalaModule {
+  def scalaVersion = "3.7.3"
+  
+  def ivyDeps = Agg(
+    mvn"it.unibo.scafi::scafi3-core:1.0.5",
+    // For distributed systems support
+    mvn"it.unibo.scafi::scafi3-distributed:1.0.5"
+  )
+}
+```
+
+For cross-platform projects with Mill:
+
+```scala
+import mill._, scalalib._, scalajslib._, scalanativelib._
+
+object myproject extends Cross[MyProjectModule](JVMPlatform, JSPlatform, NativePlatform)
+
+class MyProjectModule(val platform: Platform) extends CrossPlatformScalaModule {
+  def scalaVersion = "3.7.3"
+  
+  def ivyDeps = Agg(
+    mvn"it.unibo.scafi::scafi3-core:1.0.5",
+  )
+}
+```
+
+## Quick Start
+
+### Basic Field Operations
+
+The foundation of ScaFi is the **field calculus**, which provides three core primitives:
+
+```scala
+import it.unibo.scafi.language.xc.ExchangeLanguage
+import it.unibo.scafi.language.xc.calculus.ExchangeCalculus
+import it.unibo.scafi.libraries.All.*
+
+// Define your aggregate program using context functions
+def myProgram(using context: ExchangeCalculus & ExchangeLanguage): Int =
+  // neighborValues: share a value with neighbors and get their values back
+  val neighborIds = neighborValues(localId)
+  
+  // evolve: maintain state across rounds (like a distributed variable)
+  val roundCount = evolve(0)(count => count + 1)
+  
+  // share: compute values while sharing them with neighbors
+  val consensus = share(0)(neighborValues => 
```

Comment:
I would not call this "consensus". It will a different value for each node. Or we should change that with a "max" aggregate, otherwise I think it will not converge to a single value